internal/testing/oteltest/exporter.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2019-2025 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package oteltest supports testing of OpenTelemetry integrations.
+package oteltest
+
+import (
+	"context"
+	iotel "gocloud.dev/internal/otel"
+	"sync"
+	"testing"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	noopmetric "go.opentelemetry.io/otel/metric/noop"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+)
+
+// TestExporter is an exporter of OpenTelemetry traces and metrics, for testing.
+// It should be created with NewTestExporter.
+type TestExporter struct {
+	mu             sync.Mutex
+	spanExporter   *tracetest.InMemoryExporter
+	metricExporter *MetricExporter
+	shutdown       func(context.Context) error
+}
+
+// MetricExporter is a simple metrics exporter for testing.
+type MetricExporter struct {
```

Comment:
It looks like you have some functions implemented on this struct because you want it to implement some kind of interface? If that's the case, please note that in the comment.